cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/TimePartitionedFileSetArguments.java

```diff
@@ -54,6 +55,32 @@ public static Long getOutputPartitionTime(Map<String, String> arguments) {
   }
 
   /**
+   * This is the file extension for each partition when using TimePartitionedFileSet as an OutputFormatProvider.
+   * It's used to generate the end of the output file path for each partition.
+   * @param pathFormat The format for the path; for example: '%tF/%tH-%tM.%d',
+   *                   which will create a file path ending in the format of 2015-01-01/20-42.142017372000
+   *                   The pathFormat will be the format provided to
+   *                   {@link String}.format(String format, Object ... args), and args will be the appropriate
+   *                   number of instances of the time in milliseconds since epoch time (00:00:00 January 1, 1970 UTC).
+   */
+  public static void setOutputPathFormat(Map<String, String> arguments, String pathFormat) {
+    arguments.put(OUTPUT_PATH_FORMAT, pathFormat);
```

Comment:
should get that the format is a valid format and throw an exception if so.  String.format already throws IllegalFormatException